### PR TITLE
Resizable Grid Columns

### LINKED ISSUE #935: Grid: Resizable Columns
**Enhancement**

Add support for resizable columns in the grid. The opt-in feature will make all columns resizable and  columns sizes will not be saved across instances of the grid.

### PR DESCRIPTION
**Pull request by agubler**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [x] Unit tests are included in the PR
* [ ] For new widgets, an entry has been added to the `.dojorc`
* [ ] Any widget variant uses `theme.compose` like [this](https://github.com/dojo/widgets/issues/847)
* [ ] WidgetProperties are exported

**Description:**

Adds support for resizable grid columns via a new optional property in `ColumnConfig`. Setting column as `resizable` will enable the column to be resized via clicking the right side column border and dragging left or right.

```ts
const columnConfig = [
    {
          id: 'id',
          title: 'Id',
          resizable: true
    },
    {
          id: 'name',
          title: 'Name'
    },
    {
          id: 'id',
          title: 'Id',
          resizable: true
    }
];
```

**Note:** The column widths are not persistent across instances of the grid and cannot be set externally to the grid.

Resolves #935

### CODE REVIEW

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-jcf1m25vb.now.sh
* Tests Deployment: https://dojowidgets-ankjcdmjh.now.sh

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-mnb3nt2xj.now.sh
* Tests Deployment: https://dojowidgets-3d4idbgus.now.sh

File: src/grid/styles/header.m.css [hunk 1]
```diff
@@ -1,7 +1,27 @@
...
+	position: absolute;
+	right: -3px;
+	width: 5px;
+	top: 0;
+	bottom: 0;
+	cursor: col-resize;
+	z-index: 1;
+}
+
+.resize:hover {
```
Comment (tomdye) [inline src/grid/styles/header.m.css]:
```suggestion
.resize:hover, .resize:active {
```

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-gjzilcszv.now.sh
* Tests Deployment: https://dojowidgets-jxarxcb1a.now.sh

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-mvxaqje5x.now.sh
* Tests Deployment: https://dojowidgets-3mhi4z2gf.now.sh

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-5jinaesbw.now.sh
* Tests Deployment: https://dojowidgets-iugw5lv48.now.sh

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-5jinaesbw.now.sh
* Tests Deployment: https://dojowidgets-fya4paihd.now.sh

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-5jinaesbw.now.sh
* Tests Deployment: https://dojowidgets-rdi01z8yv.now.sh

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-3jb0emud2.now.sh
* Tests Deployment: https://dojowidgets-bc3k4i4yg.now.sh

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-ekn8j7c5e.now.sh
* Tests Deployment: https://dojowidgets-deo4lm253.now.sh

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-pnxujfy98.now.sh
* Tests Deployment: https://dojowidgets-72lwcw5x6.now.sh

File: src/grid/Cell.tsx [hunk 2]
```diff
@@ -86,40 +88,52 @@ export default class Cell extends ThemedMixin(FocusMixin(WidgetBase))<CellProper
...
-							focus: this._focusKey === 'button' ? this.shouldFocus : () => false,
-							type: 'button',
-							extraClasses: { root: this.theme(css.edit) } as any,
-							onClick: this._onEdit
-						},
-						[w(Icon, { type: 'editIcon', altText: 'Edit', classes, theme })]
-				  )
-				: null
-		]);
+							label: `Edit ${rawValue}`,
```
Comment (nicknisi) [inline src/grid/Cell.tsx]: Maybe this would belong in a separate issue, but these labels should probably use i18n.

File: src/grid/index.tsx [hunk 3]
```diff
@@ -82,11 +86,38 @@ export default class Grid<S> extends ThemedMixin(WidgetBase)<GridProperties<S>>
...
+		const currentColumnWidth = this._columnWidths[columnConfig[index].id];
+		if (currentColumnWidth <= MIN_COLUMN_WIDTH && value < 0) {
+			return;
+		}
+
+		if (currentColumnWidth + value <= MIN_COLUMN_WIDTH) {
+			value = value - (currentColumnWidth + value - MIN_COLUMN_WIDTH);
+		}
+
+		// this._rowWidth = this._rowWidth + value;
```
Comment (nicknisi) [inline src/grid/index.tsx]: Remove commented out line.
```suggestion
```

File: src/grid/index.tsx [hunk 4]
```diff
@@ -115,6 +146,7 @@ export default class Grid<S> extends ThemedMixin(WidgetBase)<GridProperties<S>>
 	}
 
 	private _onScroll(value: number) {
+		console.log('here');
```
Comment (nicknisi) [inline src/grid/index.tsx]: Remove console.log.
```suggestion
```

File: src/grid/Cell.tsx (hunk 2 above)
Comment (agubler) [reply src/grid/Cell.tsx]: Yup that should be a separate issue. I've not added this as part of these changes, just the indentation has changed.

**Comment by agubler** [comment]
Thanks @nicknisi, comments addressed.

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-9wuzy5sks.now.sh
* Tests Deployment: https://dojowidgets-ohnnzse9b.now.sh

**Pull request merged**